-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48041/44786

ERROR: Build errors found during clang-tidy run.
```
src/RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsAlgoPortable.dev.cc:40:2: error: #endif without #if [clang-diagnostic-error]
   40 | #endif
      |  ^
Suppressed 704 warnings (704 in non-user code).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsAlgoPortable.dev.cc
+++ b/RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsAlgoPortable.dev.cc
@@ -34,7 +34,11 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::ecal::weights {
       //divide the grid into uniform elements
       for (auto tx : uniform_elements(acc, nchannels)) {
         bool g1 = false;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wdangling-reference"
         const auto& digi = digisDev[tx].data();
+#pragma GCC diagnostic pop
         auto recHit = uncalibratedRecHitsDev[tx];
         recHit.amplitude() = 0;
         recHit.jitter() = 0;

